### PR TITLE
Create doc for common attributes to most/all AMP components

### PR DESCRIPTION
**Pull request by bpaduch**
A doc that will be linked/referred from amp component docs that describe the common attributes for AMP components.

to: @pbakaus 
cc: @cramforce 

In a different PR, I'll show how an amp component's doc "attributes" section will refer to this doc. Example - https://github.com/ampproject/amphtml/pull/6772

### CODE REVIEW

File: content/docs/reference/common_attributes.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    src="/img/cat.gif"
+    alt="cat animation">
+</amp-anim>
+[/sourcecode]
+
+For more information, see [Layout & Media queries](https://www.ampproject.org/docs/guides/responsive/control_layout) and the [Layout Spec](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-layout.md).
+
+## heights
+
+All AMP elements that support the `responsive` layout, also support the `heights` attribute. The value of this attribute is a sizes expression based on media expressions, similar to the [img sizes attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img) but with two key differences:
```
Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: I'd reword as: "Similar to the sizes attribute on &lt;img&gt; tags"

File: content/docs/reference/common_attributes.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+
+ 1. The value applies to the height, not the width of the element.
+ 2. Percent values are allowed. A percent value indicates the percent of the element's width. For example, a value of `80%` indicates that the height of the element will be 80% of the element's width.
+
+When the `heights` attribute is specified along with `width` and `height`, the `layout` is defaulted to `responsive`.
```
Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: This could be a note.

File: content/docs/reference/common_attributes.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+Example: 
+
+[sourcecode:html]
+<amp-img src="amp.png"
+    width="320" height="256"
+    heights="(min-width:500px) 200px, 80%">
+</amp-img>
+[/sourcecode]
+
+For more information, see [Art direction with srcset, sizes & heights](https://www.ampproject.org/docs/guides/responsive/art_direction).
```
Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: I usually styled these as callouts (see styling guides), but up to you :) I'm torn, would like a better, more distinct way of telling people there's a continuation.

File: content/docs/reference/common_attributes.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    width="466"
+    height="355" layout="responsive"></amp-img>
+<amp-img
+    media="(max-width: 649px)"
+    src="narrow.jpg"
+    width="527"
+    height="193" layout="responsive"></amp-img>
+[/sourcecode]
+
+For more information, see [Layout & Media queries](https://www.ampproject.org/docs/guides/responsive/control_layout).
```
Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: Link to the actual section, e.g. add #element-media-queries to the link.

File: content/docs/reference/common_attributes.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+Example:
+
+[sourcecode:html]
+<button on="tap:my-lightbox">Open lightbox</button>
+<amp-lightbox id="my-lightbox" layout="nodisplay">
+  ...
+</amp-lightbox>
+[/sourcecode]
+
+For more information, see  [Actions and Events in AMP](https://github.com/ampproject/amphtml/blob/master/spec/amp-actions-and-events.md).
```
Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: Would be good to migrate this doc into a guide, e.g. "Adding interactions" or something. Just a reminder, no blocker for this PR.

File: content/docs/reference/common_attributes.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+* [media](#media)
+* [noloading](#noloading)
+* [on](#on)
+* [placeholder](#placeholder)
+* [sizes](#sizes)
+* [width](#width)
+
+## fallback
+
+The `fallback` attribute can be set on any HTML element, not just AMP elements. A fallback is a convention that allows the element to communicate to the reader that the browser does not support the element. If specified, a fallback element must be a direct child of the AMP element. The exact behavior with respect to the fallback is up to the element's implementation.
```
Comment (cramforce) [inline content/docs/reference/common_attributes.md]: … does not support the element or that loading the underlying resource failed.

Comment (cramforce) [inline content/docs/reference/common_attributes.md]: to the end:, but typically the fallback element would be shown in place of the regular element.

File: content/docs/reference/common_attributes.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+[sourcecode:html]
+<amp-anim src="animated.gif" width="466" height="355" layout="responsive" >
+  <div fallback>Cannot play animated images on this device.</div>
+</amp-anim>
+[/sourcecode]
+
+For more information, see [Placeholders & fallbacks](https://www.ampproject.org/docs/guides/responsive/placeholders).
+
+## height
```
Comment (cramforce) [inline content/docs/reference/common_attributes.md]: Maybe make width and height one paragraph? It is a bit weird to have them far apart.

File: content/docs/reference/common_attributes.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+    src="narrow.jpg"
+    width="527"
+    height="193" layout="responsive"></amp-img>
+[/sourcecode]
+
+For more information, see [Layout & Media queries](https://www.ampproject.org/docs/guides/responsive/control_layout).
+
+## noloading
+
+The `noloading` attribute indicates whether the "loading indicator" should be turned **off** for this element. Many AMP elements are white-listed to show a "loading indicator", which is a basic animation that shows that the element has not yet fully loaded.
```
Comment (cramforce) [inline content/docs/reference/common_attributes.md]: Remove `are white-listed to`

File: content/docs/reference/common_attributes.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+<amp-lightbox id="my-lightbox" layout="nodisplay">
+  ...
+</amp-lightbox>
+[/sourcecode]
+
+For more information, see  [Actions and Events in AMP](https://github.com/ampproject/amphtml/blob/master/spec/amp-actions-and-events.md).
+
+## placeholder
+
+The `placeholder` attribute can be set on any HTML element, not just AMP elements. The placeholder attribute indicates that the element marked with this attribute acts as a placeholder for the parent AMP element. If specified, a placeholder element must be a direct child of the AMP element.
```
Comment (cramforce) [inline content/docs/reference/common_attributes.md]: The first sentence here and for `fallback` is a bit misleading.

Maybe lead with what they are for and then say "The attribute can be placed on any HTML element that is a direct child of an amp element that supports placeholders."

File: content/docs/reference/common_attributes.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+By default, the placeholder is immediately shown for the AMP element, even if the AMP element's resources have not been downloaded or initialized. Once ready, the AMP element typically hides its placeholder and shows the content. The exact behavior with respect to the placeholder is up to the element's implementation.
```
Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: We usually use the term "component" instead of "element". I'm torn, I prefer elements but components reminds of web components, which is the underlying technology. WDYT @cramforce?

There's also the word tag, tag often gets used to describe the abstract, e.g. the -tag. Element often gets used to describe actual instances of tags on a page, e.g. "These 5  elements should not be here" (you wouldn't use tag in that scenario).

File: content/docs/reference/common_attributes.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+<amp-anim src="animated.gif" width="466" height="355" layout="responsive">
+  <amp-img placeholder src="preview.png" layout="fill"></amp-img>
+</amp-anim>
+[/sourcecode]
+
+For more information, see [Placeholders & fallbacks](https://www.ampproject.org/docs/guides/responsive/placeholders).
+
+## sizes
+
+All AMP elements that support the `responsive` layout, also support the `sizes` attribute. The value of the `sizes` attribute is a sizes expression as described in [img sizes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img), but extended to all elements, not just images.
```
Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: Same as above, would reword "img sizes" to something like the -tag's sizes attribute.

File: content/docs/reference/common_attributes.md (hunk 3 above)
Comment (bpaduch) [inline content/docs/reference/common_attributes.md]: In this case, I'm going to avoid callouts as there's many FYIs will make the doc too busy.

Comment (pbakaus) [inline content/docs/reference/common_attributes.md]: SGTM.

File: content/docs/reference/common_attributes.md (hunk 5 above)
Comment (bpaduch) [inline content/docs/reference/common_attributes.md]: Will make note (in my docs list)

File: content/docs/reference/common_attributes.md (hunk 6 above)
Comment (bpaduch) [inline content/docs/reference/common_attributes.md]: Done

File: content/docs/reference/common_attributes.md (hunk 7 above)
Comment (bpaduch) [inline content/docs/reference/common_attributes.md]: Done

File: content/docs/reference/common_attributes.md (hunk 8 above)
Comment (bpaduch) [inline content/docs/reference/common_attributes.md]: done

File: content/docs/reference/common_attributes.md (hunk 9 above)
Comment (bpaduch) [inline content/docs/reference/common_attributes.md]: done
